tests/zfs-tests/tests/functional/l2arc/l2arc_mfuonly_pos.ksh

```diff
@@ -75,7 +75,13 @@ log_must fio $FIO_SCRIPTS/random_reads.fio
 
 log_must zpool export $TESTPOOL
 log_must zpool import -d $VDIR $TESTPOOL
-log_must test $(get_arcstat l2_mru_asize) -eq 0
+
+# Regardless of l2arc_noprefetch, some MFU buffers might be evicted
+# from ARC, accessed later on as prefetches and transition to MRU as prefetches.
+# If accessed again they are counted as MRU and the l2arc_mru_asize arcstat
+# will not be 0. That is the reason with we set the following test to less
+# than 1kB and not exactly 0.
+log_must test $(get_arcstat l2_mru_asize) -le 1024
```

Comment:
I appreciate the explanation but I'm not sure it really explains where the magic 1kB comes from - does the test expect <1kB recordsize or something?

Reply:
"it's also in the man page of l2arc_mfuonly" - sad to say I'm not sure which man page this is within :(

Reply:
the man page of l2arc_mfuonly of course, in zfs-module-parameters. I will clarify.

Reply:
The magic value of 1kB is based on multiple runs. When it fails, it fails with 512 in my testing, so I doubled that value.

Reply:
Perhaps I can fix this by setting zfs_prefetch_disable=1 in the test. Let me test.

Reply:
In a local run of 1000 tests, setting zfs_prefetch_disable and restoring testing for 0 (as opposed to 1kB) produced no failures. Adopted this approach and updated the comments.